### PR TITLE
Added the exact doc string recommended in the comment

### PR DESCRIPTION
**Pull request by petezhut**
Just added the struct docstring for #233

### CODE REVIEW

**Comment by leshow** [comment]
You've got to add an internal link to get `scope()` to link properly. I recently did a bunch of these, have a look at how to do it:

https://github.com/leshow/rayon/commit/c4928f73efe15f5f4b538a518455d719d5b8575f

**Comment by petezhut** [comment]
I'll update it!

File: rayon-core/src/scope/mod.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ use unwind;
 #[cfg(test)]
 mod test;
 
+/// Represents a fork-join scope which can be used to spawn any number 
+/// of tasks. See [`Scope()`] for more information.
```
Comment (nikomatsakis) [inline rayon-core/src/scope/mod.rs]: This isn't quite right. For one thing, the name of the function is not capitalized, and you also need a link. Something like this is probably right:

    See [`scope()`][s] for more information.
    
    [s]: fn.join.html

Comment (petezhut) [inline rayon-core/src/scope/mod.rs]: Ah.  I'm very unfamiliar with whatever parsing engine you are planning to use for doc generation, so this is a brand new experience.  I'll give it another shot!

Comment (nikomatsakis) [inline rayon-core/src/scope/mod.rs]: @petezhut it's just markdown; you can check the output locally by running `cargo doc`, I think

**Pull request closed without merging**